src/webots/vrml/WbVrmlNodeUtilities.cpp

```diff
@@ -481,3 +482,40 @@ QString WbVrmlNodeUtilities::exportNodeToString(WbNode *node) {
   node->write(writer);
   return nodeString;
 }
+
+// Return true if we can convert the Transform to a Pose.
+bool WbVrmlNodeUtilities::transformBackwardCompatibility(WbTokenizer *tokenizer) {
+  if (tokenizer) {
+    const int initalIndex = tokenizer->pos();
+    bool inChildren = false;
+    int bracketCount = 0;
+    while (tokenizer->hasMoreTokens()) {
+      const QString token = tokenizer->nextWord();
+      if (inChildren) {
+        if (token == "[")
+          bracketCount++;
+        else if (token == "]") {
+          bracketCount--;
+          if (bracketCount == 0)
+            inChildren = false;
+        }
+      } else if (token == "children") {
+        inChildren = true;
+      } else if (token == "scale") {
+        for (int i = 0; i < 3; i++) {
+          if (tokenizer->nextWord() != '1') {
```

Comment:
1 may also be written "1.0". Shall we transform the token into a float and compare it to 1?